libcodechecker/check.py

```diff
@@ -0,0 +1,515 @@
+                     default=argparse.SUPPRESS,
+                     type=None,
+                     choices=None,
+                     required=False,
+                     help="(Usage of this argument is DEPRECATED and has no "
+                          "effect!)",
+                     metavar='')
+
+    def __call__(self, parser, namespace, value=None, option_string=None):
+        LOG.warning("Deprecated command line option used: '" +
+                    option_string + "'")
+
+
+def get_argparser_ctor_args():
+    """
+    This method returns a dict containing the kwargs for constructing an
+    argparse.ArgumentParser (either directly or as a subparser).
+    """
+
+    return {
+        'prog': 'CodeChecker check',
+        'formatter_class': argparse.ArgumentDefaultsHelpFormatter,
+
+        # Description is shown when the command's help is queried directly
+        'description': "Run analysis for a project with storing results "
+                       "in the database. Check only needs a build command or "
+                       "an already existing logfile and performs every step "
+                       "of doing the analysis in batch.",
+
+        # Epilogue is shown after the arguments when the help is queried
+        # directly.
+        'epilog': "If you wish to reuse the logfile resulting from executing "
+                  "the build, see 'codechecker-log'. To keep analysis "
+                  "results for later, see and use 'codechecker-analyze'. "
+                  "To store previously saved analysis results in a database, "
+                  "see 'codechecker-store'. 'CodeChecker check' exposes a "
+                  "wrapper calling these three commands in succession. Please "
+                  "make sure your build command actually builds the files -- "
+                  "it is advised to execute builds on empty trees, aka. after "
+                  "a 'make clean', as CodeChecker only analyzes files that "
+                  "had been used by the build system. Analysis results can be "
+                  "viewed by starting a 'CodeChecker server' and connecting "
+                  "to it from a Web browser, or via 'CodeChecker cmd'.",
+
+        # Help is shown when the "parent" CodeChecker command lists the
+        # individual subcommands.
+        'help': "Perform analysis on a project and store results to database."
+    }
+
+
+def add_arguments_to_parser(parser):
+    """
+    Add the subcommand's arguments to the given argparse.ArgumentParser.
+    """
+
+    # Some arguments were deprecated already in 'CodeChecker check'.
+    parser.add_argument('--keep-tmp',
+                        action=DeprecatedOptionAction)
+
+    parser.add_argument('-c', '--clean',
+                        action=DeprecatedOptionAction)
+
+    parser.add_argument('--update',
+                        action=DeprecatedOptionAction)
+
+    # TODO: --name is not required in 'store'.
```

Comment:
The `--name` argument will be kept. I meant that it's not `required`. Previously, `check` had `required=True` for `--name`, but `store` does not, as `store` _can_ be invoked without specifying `--name`, as it will _try_ to figure a name out for the run... and if it can't, it shown an error and asks the user to explicitly specify `--name`.